### ISSUE TITLE
Remove org.yaml.snakeyaml.* classes if no YAML file is used

### ISSUE THREAD

**Issue by sdeleuze** [spring-projects-experimental/spring-graal-native#28]
We should identify at build time if YAML support is needed and avoid compiling and including `org.yaml.snakeyaml.*` classes if not used. Currently these classes are always included.

To be discussed with @bclozel and @snicoll.

**Comment by sdeleuze** [comment]
As pointed out by @aclement and @bclozel, the Yaml file could frequently come externally including from K8s config maps, so let's keep it for now.

**Comment by sdeleuze** [comment]
Based on further discussion, let's keep the door open for this optimization.

I have been able to remove YAML support by using a substitution to remove `YamlJsonParser` from `JsonParserFactory` and filtering `org.springframework.boot.env.YamlPropertySourceLoader` from `spring.factories`. I then got following figures:

```
Testing commandlinerunner output
RSS memory: 27.3M (previously 29.4M)
Image size: 23.8M (previously 24.8M)
Build time: 1m25s (previously 1m11s)
```
So by removing YAML support we get a decrease of 1.1M in term of RSS memory, 1M in term of  image size and 14s in term of build time.

The hack I did is available in https://github.com/sdeleuze/spring-graal-native/commit/removeyaml in case we want to improve and integrate it.
